pkg/blobfuse-proxy/server/server.go

```diff
@@ -62,14 +63,16 @@ func (server *MountServer) MountAzureBlob(ctx context.Context,
 
 	args := req.GetMountArgs()
 	authEnv := req.GetAuthEnv()
-	klog.V(2).Infof("received mount request: Mounting with args %v \n", args)
+	protocol := req.GetProtocol()
+	klog.V(2).Infof("received mount request: Protocol: %s, Mounting with args %v \n", protocol, args)
 
 	var cmd *exec.Cmd
 	var result mount_azure_blob.MountAzureBlobResponse
-	switch server.blobfuseVersion {
-	case BlobfuseV1:
+	if server.blobfuseVersion == BlobfuseV1 && protocol == blob.Fuse {
```

Comment:
there would also be possibilities that `protocol` is empty? The logic could be changed to:
```
if protocol == fuse2 || os_version >= `22.04` {
   do blobfuse2 mount
} else {
   do blobfuse1  mount
}
```

I think this may be more safer

Reply:
I think so